src/testFixtures/java/org/opensearch/knn/KNNRestTestCase.java

```diff
@@ -121,19 +121,21 @@ public static void dumpCoverage() throws IOException, MalformedObjectNameExcepti
         }
 
         String serverUrl = System.getProperty("jmx.serviceUrl");
-        try (JMXConnector connector = JMXConnectorFactory.connect(new JMXServiceURL(serverUrl))) {
-            IProxy proxy = MBeanServerInvocationHandler.newProxyInstance(
-                connector.getMBeanServerConnection(),
-                new ObjectName("org.jacoco:type=Runtime"),
-                IProxy.class,
-                false
-            );
-
-            Path path = Path.of(jacocoBuildPath, "integTest.exec");
-            Files.write(path, proxy.getExecutionData(false));
-        } catch (Exception ex) {
-            log.error("Failed to dump coverage: ", ex);
-            throw new RuntimeException("Failed to dump coverage: " + ex);
+        if (serverUrl != null) {
```

Comment:
I think we should throw something if it is null

Reply:
Ack